lib/screenplay_web/controllers/config_controller.ex

```diff
@@ -3,18 +3,23 @@ defmodule ScreenplayWeb.ConfigController do
 
   alias Screenplay.Config.PermanentConfig
 
-  def add(conn, _params) do
-    PermanentConfig.add_new_screen()
-    conn
-  end
+  def add(conn, %{"screen_id" => screen_id, "screen" => screen, "etag" => etag}) do
+    case PermanentConfig.add_new_screen(screen_id, screen, etag) do
+      :ok ->
+        send_resp(conn, 200, "OK")
 
-  def edit(conn, _params) do
-    PermanentConfig.edit_screen()
-    conn
+      :error ->
+        send_resp(conn, 400, "Unable to add new screen")
```

Comment:
Only issue with removing the `/edit` endpoint is that this error message is a little confusing. Should this be changed to something more general like `Config is outdated`? Then again, this will mostly be used as a comment of sorts because I'm sure the client will ignore whatever goes there.

Reply:
Hmm well technically the action can also return an error if either the get or put S3 operations fail themselves. I thought about adding more specific error reasons to those cases but it seemed a bit like overkill, especially since like you said, it's likely the client won't make sure of those more specified error messages, so I just left it as a more general message communicating "that didn't work". But I could certainly still see it being helpful to be more specific here, if even for local dev. What do you think about returning error reasons and specifying the error message returned with the response here accordingly?

Reply:
I think that's a good idea. And it still keeps the code conventional by using the error tuple pattern.

Reply:
Just added some more specific error handling. Let me know if you have additional thoughts!